### DISCUSSION TITLE
Can artifacts be tampered from the outside of the workflow run associated with the artifacts?

### DISCUSSION THREAD

**Discussion by suzuki-shunsuke** [community/community#68288] [Actions] [answered]
### Select Topic Area

Question

### Body

In my understanding, artifacts uploaded by [upload-artifact](https://github.com/actions/upload-artifact) can't be tampered by outside of the workflow run associated with the artifacts.
Is this correct?

Are there any references about it?

**Comment by milindaShehan** [comment, marked as answer]
Yes, artifacts uploaded by upload-artifact cannot be tampered with outside of the workflow run associated with the artifacts.
This is because artifacts are stored in a secure location that is only accessible to the workflow run that uploaded them.

**Reply by suzuki-shunsuke** [reply]
Thank you for your answer!

I asked GitHub Support just in case, and got the same answer!

Me

> Hi, I have a question about Artifacts.
> 
> In my understanding, artifacts uploaded by the workflow run A with upload-artifact can't be tampered by outside of the workflow run A.
> Is this correct?
> 
> Are there any references about it?
> 
> I expect artifacts as a temporal secure storage, but I'd like to confirm my understanding is correct.

Support

> Hi Shunsuke,
> 
> Thank you for writing in!
> 
> I can confirm that your understanding is correct. Artifacts are stored in a secure location which is only accessible to the workflow run that uploaded them.
> 
> You can find out more about storing artifacts in our documentation here: https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts
> 
> I hope this helps.
